src/System.Text.Encoding/tests/UTF32Encoding/UTF32EncodingTests.cs

```diff
@@ -0,0 +1,154 @@
+            yield return new object[] { Encoding.GetEncoding("utf-32BE"), "utf-32BE" };
+        }
+
+        [Theory]
+        [MemberData(nameof(Encodings_TestData))]
+        public void WebName(UTF32Encoding encoding, string webName)
+        {
+            Assert.Equal(webName, encoding.WebName);
+        }
+
+        [Theory]
+        [MemberData(nameof(Encodings_TestData))]
+        public void CodePage(UTF32Encoding encoding, string webName)
+        {
+            Assert.Equal(webName == "utf-32" ? 12000 : 12001, encoding.CodePage);
+        }
+
+        [Theory]
+        [MemberData(nameof(Encodings_TestData))]
+        public void EncodingName(UTF32Encoding encoding, string _)
+        {
+            Assert.NotEmpty(encoding.EncodingName); // Unicode (UTF-32) in en-US
+        }
+
+        [Theory]
+        [MemberData(nameof(Encodings_TestData))]
+        public void IsSingleByte(UTF32Encoding encoding, string _)
+        {
+            Assert.False(encoding.IsSingleByte);
+        }
+
+        [Theory]
+        [MemberData(nameof(Encodings_TestData))]
+        public void Clone(UTF32Encoding encoding, string _)
+        {
+            UTF32Encoding clone = (UTF32Encoding)encoding.Clone();
+            Assert.NotSame(encoding, clone);
+            Assert.Equal(encoding, clone);
+        }
+
+        public static IEnumerable<object[]> Equals_TestData()
+        {
+            yield return new object[] { new UTF32Encoding(), new UTF32Encoding(false, true, false), true };
+            yield return new object[] { new UTF32Encoding(), new UTF32Encoding(false, false, false), false };
+            yield return new object[] { new UTF32Encoding(), new UTF32Encoding(true, true, false), false };
+            yield return new object[] { new UTF32Encoding(), new UTF32Encoding(false, true, true), false };
+
+            yield return new object[] { Encoding.UTF32, Encoding.UTF32, true };
+            yield return new object[] { Encoding.UTF32, new UTF32Encoding(false, true), true };
+            yield return new object[] { Encoding.UTF32, Encoding.GetEncoding("utf-32"), true };
+            yield return new object[] { Encoding.UTF32, Encoding.GetEncoding("utf-32LE"), true };
+            yield return new object[] { Encoding.UTF32, Encoding.GetEncoding("utf-32BE"), false };
+
+            yield return new object[] { Encoding.GetEncoding("utf-32BE"), new UTF32Encoding(true, true), true };
+            yield return new object[] { Encoding.GetEncoding("utf-32BE"), Encoding.GetEncoding("utf-32"), false };
+
+            yield return new object[] { new UTF32Encoding(), new object(), false };
+            yield return new object[] { new UTF32Encoding(), null, false };
+        }
+
+        [Theory]
+        [MemberData(nameof(Equals_TestData))]
+        public void Equals(UTF32Encoding encoding, object value, bool expected)
+        {
+            Assert.Equal(expected, encoding.Equals(value));
+            if (value is UTF32Encoding)
```

Comment:
Look at my comment on the PR https://github.com/dotnet/corefx/pull/8021/files
what you think?